src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1196,6 +1196,17 @@ bool isAllowedInitOnlySet(BoundExpression receiver)
                     return true;
                 }
 
+                while (receiver is BoundConversion boundConversion)
+                {
+                    var conversion = boundConversion.Conversion;
+                    if (!(conversion.IsIdentity || conversion.IsReference))
+                    {
+                        return false;
+                    }
+
+                    receiver = boundConversion.Operand;
+                }
+
                 // bad: other.InitOnlyProperty = ...
                 if (!(receiver is BoundThisReference || receiver is BoundBaseReference))
```

Comment:
>receiver is BoundBaseReference [](start = 56, length = 30)

Should we disallow conversions on top of `base`? It is illegal to convert `base` in general.

Reply:
I think it's likely fine to handle it here. We'll report an error on the conversion itself.

---
In reply to: [558459166](https://github.com/dotnet/roslyn/pull/50424#discussion_r558459166) [](ancestors = 558459166)